### PR TITLE
PC-975: Init

### PR DESCRIPTION
**Pull request by ulfryk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,58 @@
...
+import '@samwise-tech/immutable/Iterable/getMaybe';
+import { Map } from 'immutable';
+
+Map({ a: 1, b: 2, c: 3 }).getMaybe('a'); // => Some(1)
+Map({ a: 1, b: 2, c: 3 }).getMaybe('z'); // => None()
+```
+
+Depends on [`monet`][1]
+
+### Iterable getMaybe
```
Comment (tomecko) [inline README.md]: `headMaybe`?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,58 @@
...
+
+Returns a Maybe of value:
```
Comment (tomecko) [inline README.md]: :eye:

File: README.md [hunk 3]
```diff
@@ -1,2 +1,58 @@
...
+
+List([1, 2, 3]).headMaybe(); // => Some(1)
+List([]).headMaybe(); // => None()
+```
+
+Depends on [`monet`][1]
+
+### Set symdiff
+
+Returns a Maybe of value:
```
Comment (tomecko) [inline README.md]: :eye:

File: README.md [hunk 4]
```diff
@@ -1,2 +1,58 @@
...
+import '@samwise-tech/immutable/Iterable/headMaybe';
+import { List } from 'immutable';
+
+List([1, 2, 3]).headMaybe(); // => Some(1)
+List([]).headMaybe(); // => None()
+```
+
+Depends on [`monet`][1]
+
+### Set symdiff
```
Comment (tomecko) [inline README.md]: maybe rephrase so `Set` is not ambiguous (`set` as a verb vs `set` as a noun)

File: README.md [hunk 5]
```diff
@@ -1,2 +1,58 @@
...
+Returns a Maybe of value:
+
+```typescript
+import '@samwise-tech/immutable/Set/symdiff';
+import { Set } from 'immutable';
+
+Set([1, 2, 3]).symdiff(Set([2, 3, 4])); // => Set { 1, 4 }
+```
+
+## Typing fixes
```
Comment (tomecko) [inline README.md]: Typings?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,58 @@
...
+
+In internal [`ImmutableJS`][2] typings some operators (e.g. `.map`, `.filter`, `.flatMap`) called on `Set`, `List` or `Map` are typed in a way that `tsc` recognizes it's output as `Iterable` but should treat it as `Set`, `List` or `Map`. Do `import '@samwise-tech/immutable';` and `Set`, `Map` and `List` will be fixed.
```
Comment (tomecko) [inline README.md]: its output?
their output?
not "it's"

File: src/Iterable/getMaybe.spec.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+/* tslint:disable:no-magic-numbers */
+import { expect } from 'chai';
+import { Iterable, List, Map, Set } from 'immutable';
+import { None, Some } from 'monet';
+
+import './getMaybe';
+
+describe('Iterable', () => {
+  describe('.getMaybe(a) method', () => {
```
Comment (tomecko) [inline src/Iterable/getMaybe.spec.ts]: why not simply `getMaybe method`?
the way it's called (`.` and `(a)`) is not needed here

File: src/Iterable/headMaybe.spec.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  describe('.headMaybe(a) method', () => {
+
+    it('should be a function', () => {
+      expect(Iterable.prototype.headMaybe).to.be.instanceOf(Function);
+      expect(List().headMaybe).to.be.instanceOf(Function);
+      expect(Map().headMaybe).to.be.instanceOf(Function);
+      expect(Set().headMaybe).to.be.instanceOf(Function);
+    });
+
+    it('should return option of first element', () => {
```
Comment (tomecko) [inline src/Iterable/headMaybe.spec.ts]: of the first

File: src/Iterable/headMaybe.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { Iterable } from 'immutable';
+import { Maybe } from 'monet';
+
+// TODO: Add test
```
Comment (tomecko) [inline src/Iterable/headMaybe.ts]: r?

File: src/Iterable/getMaybe.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import { Iterable } from 'immutable';
+import { Maybe } from 'monet';
+
+// TODO: Add tests
```
Comment (tomecko) [inline src/Iterable/getMaybe.ts]: r?

File: src/Set/symdiff.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import { Set } from 'immutable';
+
+// ImmutableJS doesn't provide `symmetric difference`: (A ∪ B) \ (A ∩ B)
+// See: https://en.wikipedia.org/wiki/Symmetric_difference
+// TODO: Add test
```
Comment (tomecko) [inline src/Set/symdiff.ts]: :eye:

File: package.json [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+  "description": "Little extension for ImmutableJS (and typing fixes)",
+  "main": "index.js",
+  "types": "index.d.ts",
+  "scripts": {
+    "build": "rm -rf ./lib && node_modules/.bin/tsc --project tsconfig.publish.json",
+    "karma": "node_modules/.bin/karma start config/karma.conf.js",
+    "lint": "node_modules/.bin/tslint --project tsconfig.tslint.json --type-check --format verbose './src/**/*.ts'",
+    "mocha": "node_modules/.bin/mocha --opts ./mocha.opts './src/**/*.spec.ts'",
+    "prepublish": "yarn build && cp -r ./lib/* ./",
+    "test": "yarn outdated && yarn test-specs-for-only && yarn lint && yarn mocha && yarn karma && yarn build",
```
Comment (tomecko) [inline package.json]: I remember `yarn outdated` causing problems...

Comment (ulfryk) [reply package.json]: If anything is outdated… I'll keep it here so it will encourage us to make everything up to date (it's easy as it's small library).
